batcher/sweep_batch.go

```diff
@@ -0,0 +1,1109 @@
+	BatchReady bool
+
+	// Sweeps store the Sweeps that this batch currently contains.
+	Sweeps map[lntypes.Hash]Sweep
+
+	// currentHeight is the current block height.
+	currentHeight int32
+
+	// RequestChan is the channel over which new sweeps are received from.
+	RequestChan chan *Sweep
+
+	// returnChan is the channel over which sweeps are returned back to the
+	// batcher in order to enter a new batch. This occurs when some sweeps
+	// are left out because an older version of the batch tx got confirmed.
+	// This is a behavior introduced by RBF replacements.
+	returnChan chan *Sweep
+
+	// blockEpochChan is the channel over which block epoch notifications
+	// are received.
+	blockEpochChan chan int32
+
+	// spendChan is the channel over which spend notifications are received.
+	spendChan chan *chainntnfs.SpendDetail
+
+	// spendRefreshChan is a communication channel for the routines that
+	// monitor spends. This channel is simply used by the new routine to
+	// signal to the previous routine that it should exit. This helps the
+	// batch have at most one spend monitor active at a time.
+	spendRefreshChan chan int32
+
+	// confChan is the channel over which confirmation notifications are
+	// received.
+	confChan chan *chainntnfs.TxConfirmation
+
+	// confRefreshChan is a communication channel for the routines that
+	// monitor confirmations. This channel is simply used by the new
+	// routine to signal to the previous routine that it should exit. This
+	// helps the batch have at most one confirmation monitor active at a
+	//
+	confRefreshChan chan int32
+
+	// errChan is the channel over which errors are received.
+	errChan chan error
+
+	// completeChan is the channel over which the batch sends its ID back
+	// to the batcher to signal its exit.
+	completeChan chan lntypes.Hash
+
+	// batchTx is the transaction that is currently being monitored for
+	// confirmations.
+	BatchTxid *chainhash.Hash
+
+	// batchPkScript is the pkScript of the batch transaction's output.
+	BatchPkScript []byte
+
+	// RbfCache stores data related to the RBF fee bumping mechanism.
+	RbfCache RBFCache
+
+	// wallet is the wallet client used to create and publish the batch
+	// transaction.
+	wallet lndclient.WalletKitClient
+
+	// chainNotifier is the chain notifier client used to monitor the
+	// blockchain for spends and confirmations.
+	chainNotifier lndclient.ChainNotifierClient
+
+	// signerClient is the signer client used to sign the batch transaction.
+	signerClient lndclient.SignerClient
+
+	// muSig2Kit includes all the required functionality to collect
+	// and verify signatures by the swap server in order to cooperatively
+	// sweep funds.
+	muSig2SignSweep MuSig2SignSweep
+
+	// store includes all the database interactions that are needed by the
+	// batch.
+	store BatcherStore
+
+	// Cfg is the configuration for this batch.
+	Cfg *BatchConfig
+}
+
+// NewBatch creates a new batch.
+func NewBatch(cfg BatchConfig, batchTxid *chainhash.Hash,
```

Comment:
nit: could use a functional options pattern or a config here to help readability.